### PR TITLE
Addition of restrictPropagationThroughHierarchy flag in classification propagation

### PR DESCRIPTION
**Pull request by hr2904**
Via hierarchy
Via lineage

>When it comes to configure & customize these propagation paths, we only support restricting though lineage with restrictPropagationThroughLineage flag. This means propagations via hierarchy are not optional & done forcefully even when not required.

> To make it optional, we have introduced another flag which help to control propagations via hierarchy : `restrictPropagationThroughHierarchy`
> Added Argument of restrictThroughHierarchy in all methods where restrictThroughLineage was being passed, and handled the edge case of both flag being true. Added the concept of toExclude for : flag basis getting of labels.
> Handled all test cases and made fixes for them.
Added a method to test propagation config before sending it to Background task queue.

## Type of change
- [] New feature (adds functionality)

## Related issues

> Fix [#1](https://atlanhq.atlassian.net/browse/PLT-1036) 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

**Review by nikhilbonte21 — CHANGES_REQUESTED** [review]

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java [hunk 1]
```diff
@@ -1312,10 +1312,25 @@ public void createAndQueueTask(String taskType, AtlasVertex entityVertex, String
         createAndQueueTaskWithoutCheck(taskType, entityVertex, classificationVertexId, relationshipGuid);
     }
 
-    public void createAndQueueTaskWithoutCheck(String taskType, AtlasVertex entityVertex, String classificationVertexId, String relationshipGuid, Boolean currentRestrictPropagationThroughLineage) throws AtlasBaseException {
+    public void createAndQueueTask(String taskType, AtlasVertex entityVertex, String classificationVertexId, String relationshipGuid, Boolean currentRestrictPropagationThroughLineage, Boolean currentRestrictPropagationThroughHierarchy) throws AtlasBaseException {
```
Comment (nikhilbonte21) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java:1315]: This method is never called

**Review by nikhilbonte21 — CHANGES_REQUESTED** [review]

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/ClassificationAssociator.java [hunk 2]
```diff
@@ -257,6 +257,8 @@ private Map<String, List<AtlasClassification>> computeChanges(AtlasEntityHeader
             List<AtlasClassification> incomingClassifications = listOps.filter(incomingEntityHeader.getGuid(), incomingEntityHeader.getClassifications());
             List<AtlasClassification> entityClassifications = listOps.filter(entityToBeUpdated.getGuid(), entityToBeUpdated.getClassifications());
 
+            entityRetriever.verifyClassificationsPropagationMode(incomingClassifications);
```
[the remark this answers is not in the document]
Comment (hr2904) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/ClassificationAssociator.java:260]: I checked, EntityGraphRetirever is the only common place for this method to be put in.
Therefore calling it here

**Review by checkaayush — APPROVED** [review]
LGTM.

**Review by nikhilbonte21 — APPROVED** [review]
